src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticSuppressor.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    /// <summary>
+    /// The base type for diagnostic suppressors that can programmatically suppress analyzer and/or compiler non-error diagnostics.
+    /// </summary>
+    public abstract class DiagnosticSuppressor : DiagnosticAnalyzer
```

Comment:
It feels wrong that this inherits from `DiagnosticAnalyzer`. The first thing we have to do is make sure to disable most of the functionality of that class here. If we need to hold them in the same collections at some point, I think it would be more appropriate to define a marker interface for these two types and keep them separate otherwise.

Reply:
This was discussed in the analyzer design meeting and this was the chosen approach - I'll forward you the email thread.